.github/workflows/commit-workflow-all.yml

```diff
@@ -0,0 +1,369 @@
+name: commit-workflow-allinone
+
+on:
+  push:
+    branches: [ main, continuous_integration ]
+  workflow_dispatch:
+    branches: [ main, continuous_integration ]
+  pull_request:
+    branches: [ main, continuous_integration ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+
+  test-env:
+    # we will only see starts as gh filters this
+    name: test-env
+    runs-on: Ubuntu-20.04
+    steps:
+      - name: Whatever
+        run:  echo $HOME Oh man; echo $C8YUSERNAME $C8YTENANT $C8YDEVICE $C8YDEVICEID
+        env:
+          C8YUSERNAME: ${{ secrets.SECRET_C8YUSERNAME }}
+          C8YTENANT: ${{secrets.SECRET_C8YTENANT}}
+          C8YDEVICE: ${{ secrets.SECRET_C8YDEVICE_OFFSITE_ALL }}
+          C8YDEVICEID: ${{ secrets.SECRET_C8YDEVICEID_OFFSITE_ALL }}
+
+  cargo-test:
+
+    name: Run cargo test
+    runs-on: Ubuntu-20.04
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Enable cache
+        # https://github.com/marketplace/actions/rust-cache
+        uses: Swatinem/rust-cache@v1
+
+      - name: Cargo test
+        uses: actions-rs/cargo@v1
+        # https://github.com/marketplace/actions/rust-cargo
+        with:
+          command: test
+          args: --verbose
+
+# Build Workflow
+
```

Comment:
You omit other architectures than `armv7-unknown-linux-gnueabihf` because we will run all tests in RPi, right?